### ISSUE TITLE
AWS_PROVIDER not working, explicit 'provider=xx' works fine

### ISSUE THREAD

**Issue by glucas** [terraform-providers/terraform-provider-aws#8779]
Testing with terraform 0.12  and terraform-providers/aws 2.12.0, on Windows.

The following works as expected:

```
provider "aws" {
  profile = "test"
}
```

Commenting out the profile line and instead setting AWS_PROFILE to the same value in my shell does not work:

```
λ  $env:AWS_PROFILE='test'
λ  terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

------------------------------------------------------------------------

Error: No valid credential sources found for AWS Provider.
        Please see https://terraform.io/docs/providers/aws/index.html for more information on
        providing credentials for the AWS Provider
```

I can set AWS_DEFAULT_REGION in the shell and that works fine, but cannot use AWS_PROFILE.

**Comment by glucas** [comment]
Actually, this may be specific to profiles that use role_arn. Testing the same two cases above with a simple access key id configuration seems to work. But using role_arn appears to only work when I specify the profile in the AWS provider config and not using environment variables.

**Comment by bflad** [comment]
Hi @glucas 👋 Does your Terraform environment work with the `AWS_PROFILE` environment variable and `role_arn` in the AWS configuration file if the additional environment variable `AWS_SDK_LOAD_CONFIG=1` is used? e.g. `AWS_SDK_LOAD_CONFIG=1 AWS_PROFILE=test terraform plan`

This environment variable is not currently well documented in the Terraform AWS Provider documentation (there are a few open issues regarding this), but enabling this environment variable does enhance the behavior of AWS Go SDK based applications, like the Terraform AWS Provider.

**Comment by glucas** [comment]
Yes, thanks! Setting AWS_SDK_LOAD_CONFIG in my shell environment seems to work.

**Comment by bflad** [comment]
Additional information about `AWS_SDK_LOAD_CONFIG` environment variable has been merged and will publish with version 2.15.0 of the Terraform AWS Provider next week.
